common/src/main/java/org/opensearch/ml/common/connector/OciConnector.java

```diff
@@ -0,0 +1,132 @@
+import java.util.Locale;
+import java.util.Map;
+
+import static org.opensearch.ml.common.connector.ConnectorProtocols.OCI_SIGV1;
+
+/**
+ * Connector to OCI services
+ */
+@Log4j2
+@NoArgsConstructor
+@EqualsAndHashCode
+@org.opensearch.ml.common.annotation.Connector(OCI_SIGV1)
+public class OciConnector extends HttpConnector {
+    public static final String AUTH_TYPE_FIELD = "auth_type";
+
+    public static final String TENANT_ID_FIELD = "tenant_id";
+
+    public static final String USER_ID_FIELD = "user_id";
+
+    public static final String FINGERPRINT_FIELD = "fingerprint";
+
+    public static final String PEMFILE_PATH_FIELD = "pemfile_path";
+
+    public static final String REGION_FIELD = "region";
+
+    @Builder(builderMethodName = "ociConnectorBuilder")
+    public OciConnector(String name, String description, String version, String protocol,
+                        Map<String, String> parameters, Map<String, String> credential, List<ConnectorAction> actions,
+                        List<String> backendRoles, AccessMode accessMode, User owner) {
+        super(name, description, version, protocol, parameters, credential, actions, backendRoles, accessMode, owner);
+        validate();
+    }
+
+    public OciConnector(String protocol, XContentParser parser) throws IOException {
+        super(protocol, parser);
+        validate();
+    }
+
+
+    public OciConnector(StreamInput input) throws IOException {
+        super(input);
+        validate();
+    }
+
+    public OciConnector(String protocol, StreamInput input) throws IOException {
+        super(protocol, input);
+        validate();
+    }
+
+    private void validate() {
+        if (parameters == null) {
+            throw new IllegalArgumentException("Missing credential");
+        }
+        if (!parameters.containsKey(AUTH_TYPE_FIELD)) {
+            throw new IllegalArgumentException("Missing auth type");
+        }
+
+        final OciClientAuthType authType =
+                OciClientAuthType.from(
+                        parameters.get(AUTH_TYPE_FIELD).toUpperCase(Locale.ROOT));
+
+        if (authType == OciClientAuthType.USER_PRINCIPAL) {
+            if (!parameters.containsKey(TENANT_ID_FIELD)) {
+                throw new IllegalArgumentException("Missing tenant id");
+            }
+
+            if (!parameters.containsKey(USER_ID_FIELD)) {
+                throw new IllegalArgumentException("Missing user id");
+            }
+
+            if (!parameters.containsKey(FINGERPRINT_FIELD)) {
+                throw new IllegalArgumentException("Missing fingerprint");
+            }
+
+            if (!parameters.containsKey(PEMFILE_PATH_FIELD)) {
+                throw new IllegalArgumentException("Missing pemfile");
+            }
+
+            if (!parameters.containsKey(REGION_FIELD)) {
+                throw new IllegalArgumentException("Missing region");
+            }
+        }
+    }
+
+    @Override
+    public Connector cloneConnector() {
```

Comment:
add test to fix codecov warning?

Reply:
Yeah, I already had a test for cloneConnector, and cover a happy case. It is hard to cover failure case, since if we create valid connector then it is able to serialized and deserialized back to cloned connector